### PR TITLE
Trivial Followups to #1825

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Various little nits I stumbled upon in #1825. A few are actually things from #1825, but most of them are just things that happen to be in code #1825 touches.

### CODE REVIEW

File: lightning/src/chain/onchaintx.rs [hunk 1]
```diff
@@ -733,31 +733,14 @@ impl<ChannelSigner: Sign> OnchainTxHandler<ChannelSigner> {
...
 						// by us.
 						let mut are_sets_equal = true;
-						if !request.requires_external_funding() || !request.is_malleable() {
-							// If the claim does not require external funds to be allocated through
-							// additional inputs we can simply check the inputs in order as they
-							// cannot change under us.
-							if request.outpoints().len() != tx.input.len() {
+						// Do a linear search for each input (we don't expect large input sets to
+						// exist) to ensure the request's input set is fully spent to be resilient
+						// against the external claim reordering inputs.
```
Comment (ariard) [inline lightning/src/chain/onchaintx.rs:738]: I think here we can assume we have counterparty malleable transactions, where only one input is claiming a HTLC output, all the remaining ones can be freely malleable (even sounds true pre-anchor, HTLC-preimage on holder commitment transaction). Therefore, you might have to assume transactions with thousands of inputs, and the worst-case reasoning would be rather the block size. If you assume something like 41 bytes for the outpoint/nsequence/script_sig and 1 WU for the witnesscript (let's say an OP_TRUE), you might have each input taking 165 WU, and for a block ~24k of inputs to iter on ? So inner iterations would be more like 576M. Even if this reasoning is more accurate, and you could really stale a LDK node on this, I think this a low-concern as your counterparty would have to buy the whole blockspace, and keep going on.

**Review by ariard — COMMENTED** [review]
There is also this comment from #1825, if we would like to have sane variable names: https://github.com/lightningdevkit/rust-lightning/pull/1825#discussion_r1040245907

File: lightning/src/chain/onchaintx.rs (hunk 1 above)
Comment (ariard) [reply lightning/src/chain/onchaintx.rs]: From my understanding, if the iteration is based on the transaction matched inputs, the counterparty could influence it. Though, as mentioned I don't think this is a real concern as we're ultimately bounded by the block size here. Effectively, not so bad to be more conservative here, if we modify our aggregation logic to enable cross-channel one, we could have more than 483*2 input in a transaction issued by ourselves. We might do so in the future to save on fees.

**Review by ariard — COMMENTED** [review]
ACK 67f9f01

Thanks for clarifying the comment, this is really valuable in the future to document all our aggregation and claiming assumptions, if we enable cross-channel aggregation for fee/weight compression performance reasons (even if it's years from now).

**Comment by ariard** [comment]
> Sure, feel free to open a followup. Neither this PR nor 1825 were touching that code.

I'll take it on its own PR, addressing few more followups from #1825, especially starting to document the external fee-bumping reserves requirements in vanilla anchor output deployment.

**Review by dunxen — APPROVED** [review]

**Pull request merged**